server/service/appconfig.go

```diff
@@ -656,6 +656,8 @@ func (svc *Service) validateMDM(
 		mdm.MacOSUpdates.Deadline != oldMdm.MacOSUpdates.Deadline
 
 	if updatingVersion || updatingDeadline {
+		// TODO: Should we validate MDM configured on here too?
```

Comment:
I think so? It looks to me like it should have the same validations as other mdm-only settings. That being said, I can see that being addressed later on and even in a distinct ticket, so that this PR can land faster and fix the main issue.